scripts/zones/Aht_Urhgan_Whitegate/npcs/Lathuya.lua

```diff
@@ -9,36 +9,129 @@ require("scripts/globals/quests")
+-- itemBit = {firstItem, secondItem, thirdItem, fourthItem, coinType, coinAmmount, craftedItem, optionId}
+    [1] = {754,  828,  879, 4158, 2186, 2, 14928, 1}, -- magus bazubands
+    [2] = {761,  828, 2175, 2340, 2186, 2, 15600, 2}, -- magus shalwar
+    [4] = {828, 2229, 2288, 2340, 2186, 4, 14521, 3}  -- magus jubbah
+}
+
 function onTrade(player, npc, trade)
+    local currentTask = player:getCharVar("LathuyaCurrentTask_Mask")
+    local craftingBluArmor = player:getCharVar("CraftingBluArmor")
+    local totalCraftedPieces = player:countMaskBits(player:getCharVar("LathuyaCraftingList_Mask"))
+    local csOffset = 8 * totalCraftedPieces
+
+    if craftingBluArmor == 0  and npcUtil.tradeHasExactly(trade, {
+        craftingItems[currentTask][1], craftingItems[currentTask][2], craftingItems[currentTask][3], craftingItems[currentTask][4]
+    }) then
+        player:startEvent(732 + csOffset, craftingItems[currentTask][7], craftingItems[currentTask][5], craftingItems[currentTask][6])
+    elseif craftingBluArmor == 1 and npcUtil.tradeHasExactly(trade, {
+        {craftingItems[currentTask][5], craftingItems[currentTask][6]}
+    }) then
+        player:startEvent(734 + csOffset, 0, craftingItems[currentTask][5], craftingItems[currentTask][6])
+   end
 end
 
 function onTrigger(player, npc)
     local omensProgress = player:getCharVar("OmensProgress")
     local omens = player:getQuestStatus(AHT_URHGAN, tpz.quest.id.ahtUrhgan.OMENS)
+    local transformations = player:getQuestStatus(AHT_URHGAN, tpz.quest.id.ahtUrhgan.TRANSFORMATIONS)
+    local craftingBluArmor = player:getCharVar("CraftingBluArmor")
+    local craftedPieces = player:getCharVar("LathuyaCraftingList_Mask")
+    local totalCraftedPieces = player:countMaskBits(player:getCharVar("LathuyaCraftingList_Mask"))
+    local currentTask = player:getCharVar("LathuyaCurrentTask_Mask")
+    local pickupReady = vanaDay() > player:getCharVar("BluPaymentDay")
+    local craftmanAwake = vanaDay() > player:getCharVar("BluRestingDay")
+    local csOffset = 8 * totalCraftedPieces
 
     -- OMENS
-    if omens >= QUEST_ACCEPTED then
+    if omens == QUEST_ACCEPTED then
         if omensProgress == 3 then
             player:startEvent(714) -- Tells Master location
         elseif omensProgress == 4 then
             player:startEvent(715) -- Reminder of master location
         elseif omensProgress == 5 then
             player:startEvent(716) -- Master spoken to
-        elseif omens == QUEST_COMPLETED then
-            player:startEvent(771) -- Default dialog
         end
 
+    -- CRAFTING OTHER 3 BLUE MAGE ARMOR PIECES
+    elseif transformations >= QUEST_ACCEPTED then
+        if currentTask == 0 and totalCraftedPieces ~= 3 then
+            if craftmanAwake then
+                if totalCraftedPieces == 2 then
+                    currentTask = 7 - craftedPieces
+                    player:startEvent(746, 0, 0, 0, 0, 0, 0, 0, craftingItems[currentTask][8]) -- 8th param is response
```

Comment:
Won't this always result in an 8th param of 3 (same as the Jubbah)?

Reply:
no cause they are not completed in order, if you complete the the 1st and 2nd item then it will result in 7- 1 - 2 = 4
if you complete the 1st and 3rd item then it will be 7 - 1 - 4 =  2
if you complete the 2nd and 3rd it will be 7 - 2 - 4 = 1
then depending on that it will pass the 1st 2nd or 3rd options (sorry i don't make sense explaining here but we can get on a video call and i can explain better)

Reply:
Oh, because `craftedPieces` is the pieces crafted, not `totalCraftedPieces`.

I was reading the second line as `currentTask = 7 - totalCraftedPieces`